app/preprints/-components/submit/preprint-state-machine/component.ts

```diff
@@ -221,11 +235,62 @@ export default class PreprintStateMachine extends Component<StateMachineArgs>{
     public async onSubmit(): Promise<void> {
         this.args.resetPageDirty();
 
+        if (this.isNewVersionFlow) {
+            try {
+                const url = this.preprint.links.preprint_versions as string;
+                if (url && this.tempVersion) {
+                    const savedVersionData = await this.currentUser.authenticatedAJAX({
+                        url,
+                        type: 'POST',
+                        headers: {
+                            'Content-Type': 'application/json',
+                        },
+                        data: JSON.stringify({
+                            data: {
+                                type: 'preprints',
+                                attributes: {
+                                    primary_file: (await this.tempVersion.primaryFile)?.get('id'),
+                                },
+                            },
+                        }),
+                    });
+                    this.store.pushPayload('preprint', savedVersionData);
+                    const storedPreprintRecord = this.store.peekRecord('preprint', savedVersionData.data.id);
+                    let toastMessage = this.intl.t('preprints.submit.new-version.success');
+
+                    if (this.provider.reviewsWorkflow) {
+                        toastMessage = this.intl.t('preprints.submit.new-version.success-review');
+                        const reviewAction = this.store.createRecord('review-action', {
+                            actionTrigger: ReviewActionTrigger.Submit,
+                            target: storedPreprintRecord,
+                        });
+                        await reviewAction.save();
+                    } else {
+                        storedPreprintRecord.isPublished = true;
+                        await storedPreprintRecord.save();
+                    }
+                    this.tempVersion.destroyRecord();
+                    await this.preprint.reload(); // Refresh the original preprint as this is no longer latest version
+                    this.toast.success(toastMessage);
+                    this.router.transitionTo('preprints.detail', this.provider.id, storedPreprintRecord.id);
+                }
+            } catch (e) {
+                const errorTitle = this.intl.t('preprints.submit.new-version.error.title');
+                let errorMessage = this.intl.t('preprints.submit.new-version.error.description',
+                    { preprintWord: this.provider.documentType.singular });
+                if (e.errors[0].status === 409) { // Conflict
```

Comment:
@cslzchen Are there other status codes that we should anticipate?

Reply:
I think this is the only one for now.

(Per API contract, 401/403 and 404s is for when requests was made directly to API without via FE.)